src/coreclr/src/jit/morph.cpp

```diff
@@ -7527,11 +7527,11 @@ GenTree* Compiler::fgMorphPotentialTailCall(GenTreeCall* call)
             {
                 callType = TYP_I_IMPL;
             }
-            else if (howToReturnStruct == SPK_ByValueAsHfa)
+            else if (howToReturnStruct == SPK_ByValueAsHfa || varTypeIsSIMD(callType))
```

Comment:
I'm not terribly familiar with this code, but it seems that it is not attempting to actually provide a valid return value, but ensuring that there will not be downstream errors - is that right?

Reply:
Yes, it is absolutely right.  at that moment the original call is placed as the root node in the current statement, call's type is changed to void, all that we want to do - exit morphing of any parent nodes that are on the stack without errors. 

The parent nodes could be different - assignments, returns, casts.I did not see any errors when I was just returning TYP_INT 0 node always, but I can't prove that it is safe in general.

Reply:
Thanks - it might be worth it to add a comment to the effect.